src/annotator.cpp

```diff
@@ -252,108 +435,80 @@ void Annotator::buildModelIndex(const ModelPtr &model)
+    issue->setDescription("This Annotator object does not have a model to work with.");
+    issue->setLevel(Issue::Level::ERROR);
     mAnnotator->addIssue(issue);
-    return issue;
 }
 
-IssuePtr Annotator::AnnotatorImpl::makeIssueNoModel() const
+void Annotator::AnnotatorImpl::addInvalidArgument(CellMLElement type) const
 {
     auto issue = Issue::create();
-    issue->setDescription("No model is stored; no action has been taken.");
+    issue->setDescription("The given type '" + typeAsString(type) + "' is invalid for this method.");
     issue->setLevel(Issue::Level::ERROR);
     mAnnotator->addIssue(issue);
-    return issue;
 }
 
-AnyItem Annotator::item(const std::string &id)
+bool Annotator::AnnotatorImpl::exists(const std::string &id, size_t index) const
 {
-    AnyItem i;
-    if (!isBuilt()) {
-        return std::make_pair(Annotator::Type::ISSUE, mPimpl->makeIssueIsNotBuilt());
-    }
-    if (mPimpl->mIdList.empty()) {
-        // Setting the error message to be the same as below.
-        return std::make_pair(Annotator::Type::ISSUE, mPimpl->makeIssueNotFound(id));
+    bool exists = true;
+    if (!mAnnotator->hasModel()) {
+        addIssueNoModel();
+        exists = false;
+    } else if (mIdList.count(id) <= index || mIdList.count(id) == 0) {
+        addIssueNotFound(id);
+        exists = false;
     }
 
-    // Check whether the id has been duplicated in the model.
-    if (!isUnique(id, true)) {
-        // Get the issue created inside the isUnique() function and return it.
-        auto c = issue(issueCount() - 1);
-        i = std::make_pair(Annotator::Type::ISSUE, c);
-        return i;
+    return exists;
+}
+
+AnyItem Annotator::item(const std::string &id)
+{
+    auto retrieved = item(id, 0);
+    if (retrieved.first != CellMLElement::UNDEFINED && !isUnique(id)) {
+        mPimpl->addIssueNonUnique(id);
     }
-    return item(id, 0);
+    return retrieved;
 }
 
 AnyItem Annotator::item(const std::string &id, size_t index)
 {
-    // Retrieve the item from the idList.
-    if (index >= mPimpl->mIdList.count(id)) {
-        auto i = std::make_pair(Annotator::Type::ISSUE, mPimpl->makeIssueOutOfRange(id, index));
-        // Adding the requested id string and this issue to the map, so that future requests (ie: to the .second)
-        // will return the same IssuePtr rather than creating a new one.
-        mPimpl->mIdList.insert(std::make_pair(id, i));
-        return std::move(i);
-    }
-    return std::move(items(id)[index]);
+    return mPimpl->exists(id, index) ? std::move(items(id)[index]) : std::make_pair(CellMLElement::UNDEFINED, nullptr);
 }
 
-bool Annotator::isUnique(const std::string &id, bool raiseError = false)
+bool Annotator::isUnique(const std::string &id)
 {
-    if (mPimpl->mIdList.count(id) > 1) {
-        if (raiseError) {
-            mPimpl->makeIssueNonUnique(id);
-        }
-        return false;
-    }
-    return true;
+    return mPimpl->mIdList.count(id) == 1;
```

Comment:
This needs to go back to the `> 1` test rather than `== 1`.  Otherwise we get a return of `false` when the id doesn't exist, which doesn't make sense...